File: Java - Old/app/src/main/java/com/sathwikd/pdftoimage/PDFtoImageThread.java
package com.sathwikd.pdftoimage;

import android.app.Activity;
import android.graphics.Bitmap;
import android.os.Handler;
import android.os.ParcelFileDescriptor;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;

public class PDFtoImageThread implements Runnable {
    private final Activity activity;
    private final ProgressBar progressBar;
    private final Handler handler;
    private final ParcelFileDescriptor pfd;
    private final String fileName;
    private final int quality;
    private final Bitmap.CompressFormat formatImage;
    //private final Context context;

    public PDFtoImageThread(Handler handler, ParcelFileDescriptor pfd, String fileName, int quality, Bitmap.CompressFormat format, ProgressBar progressBar, Activity activity) {
        this.handler = handler;
        this.pfd = pfd;
        this.fileName = fileName;
        this.quality = quality;
        this.formatImage = format;
        //this.context = context;
        this.progressBar = progressBar;
        this.activity = activity;
    }

    @Override
    public void run() {
        activity.runOnUiThread(() -> progressBar.setVisibility(View.VISIBLE));
        String message = PdfUtilsSat.savePDFasImage(pfd, fileName, quality, formatImage);
        handler.post(() -> Toast.makeText(activity, message, Toast.LENGTH_SHORT).show());
        activity.runOnUiThread(() -> progressBar.setVisibility(View.INVISIBLE));
    }
}

File: Kotlin/app/src/main/java/com/sathwikd/pdftoimage/PDFtoImageThread.kt
package com.sathwikd.pdftoimage

import android.app.Activity
import android.graphics.Bitmap
import android.os.Handler
import android.os.ParcelFileDescriptor
import android.view.View
import android.widget.ProgressBar
import android.widget.Toast

class PDFtoImageThread(
    private val handler: Handler,
    private val pfd: ParcelFileDescriptor,
    private val fileName: String,
    private val quality: Int,
    private val formatImage: Bitmap.CompressFormat,
    private val progressBar: ProgressBar,
    private val activity: Activity
) : Runnable, PdfConversionProgressListener {

    override fun run() {
        activity.runOnUiThread {
            progressBar.visibility = View.VISIBLE
            progressBar.max = 100 // Default value, will be updated in onMaxProgressDetermined
        }
        val message = PdfUtilsSat.savePDFasImage(pfd, fileName, quality, formatImage, this)
        handler.post { Toast.makeText(activity, message, Toast.LENGTH_SHORT).show() }
        activity.runOnUiThread { progressBar.visibility = View.INVISIBLE }
    }

    override fun onProgressUpdate(pageNumber: Int) {
        activity.runOnUiThread {
            progressBar.progress = pageNumber
        }
    }

    override fun onMaxProgressDetermined(maxProgress: Int) {
        activity.runOnUiThread {
            progressBar.max = maxProgress
        }
    }
}